### PR TITLE
Resolving relative paths from the configuration file #36

### PR DESCRIPTION
**Pull request by ZanSara**
Targets issue #36 

Adds the function `resolvePath()` to replace the previously used `os.path.isabs()`, as the latter uses `os.getcwd()` as reference to resolve relative paths, which is incorrect if the script is launched by any other folder that the one containing the configuration file.

However, `resolvePath()` does not strictly require the configuration file path, but can receive any base path as input. For now, if a configuration file was not given, I pass `os.getcwd()` to simulate the old behavior. I wonder whether passing the path to the script (so `os.path.dirname(os.path.realpath(__file__))`) would be more reasonable, but this can be discussed in a separate issue.

`resolvePath()` also detects and returns absolute paths unmodified.

### CODE REVIEW

**Comment by ZanSara** [comment]
Sorry for the eleven extra commits, they are irrelevant. Only the last actually intriduces any modification.

Also tests and a review of the readme are coming.

**Comment by ZanSara** [comment]
I would say I'm done for now. By the way, please let's try to merge this asap because it's close to critical for my specific usecase, and I wouldn't like to have to rely on my branch.

File: pyqt5ac.py [hunk 1]
```diff
@@ -143,6 +143,24 @@ def replaceVariables(variables_definition, string_with_variables):
     return string_with_variables
 
 
+def resolvePath(path: str, reference_path: str) -> str:
```
Comment (zkovari) [inline pyqt5ac.py:146]: should be internal

File: pyqt5ac.py [hunk 2]
```diff
@@ -143,6 +143,24 @@ def replaceVariables(variables_definition, string_with_variables):
...
+    Translates relative paths into absolute paths, using the reference path as base.
+    Meaningful reference values for the caller might be the configuration file path, the script's path or the current
+    working directory.
+    :param path: path to resolve.
+    :param reference_path: path to be used as a reference to resolve absolute paths
+    :return: an absolute path corresponding to the relative path passed in input if it was relative, or the unchanged
+    input if it was an absolute path.
+    :raises: ValueError if the reference path is not absolute
+    """
+    if not os.path.isabs(path):
```
Comment (zkovari) [inline pyqt5ac.py:157]: rather
```
if os.path.isabs(path):
    return path
```
and then you can avoid the nested conditions. but ok I'll manicure this a little bit

File: pyqt5ac.py [hunk 3]
```diff
@@ -177,6 +195,9 @@ def main(rccOptions='', uicOptions='', force=False, config='', ioPaths=(), varia
         # Replace instances of the variables with the actual values of the available variables
         sourceFileExpr = replaceVariables(variables, sourceFileExpr)
 
+        # Retrieve the absolute path to the source files
+        sourceFileExpr = resolvePath(sourceFileExpr, (os.path.dirname(config) or os.getcwd()))
```
Comment (zkovari) [inline pyqt5ac.py:199]: this will keep the glob wildcards, so that's okay

File: pyqt5ac.py [hunk 4]
```diff
@@ -177,6 +195,9 @@ def main(rccOptions='', uicOptions='', force=False, config='', ioPaths=(), varia
...
+
         # Find files that match the source filename expression given
         for sourceFilename in glob.glob(sourceFileExpr, recursive=True):
```
Comment (zkovari) [inline pyqt5ac.py:202]: and here the sourceFilename will be indeed absolute, so it's okay

**Comment by zkovari** [comment]
cwd is good as an alternative. Using the script's location would be a mistake IMO

**Pull request merged**